util/compare_test_results.py

```diff
@@ -0,0 +1,204 @@
+    reference_flat = flatten_test_results(reference_results)
+
+    # Identify different categories of test changes
+    regressions, fixes, newly_skipped, newly_passing = identify_test_changes(
+        current_flat, reference_flat
+    )
+
+    # Filter out intermittent issues from regressions
+    real_regressions = [r for r in regressions if r not in ignore_list]
+    intermittent_regressions = [r for r in regressions if r in ignore_list]
+
+    # Print summary stats
+    print(f"Total tests in current run: {len(current_flat)}")
+    print(f"Total tests in reference: {len(reference_flat)}")
+    print(f"New regressions: {len(real_regressions)}")
+    print(f"Intermittent regressions: {len(intermittent_regressions)}")
+    print(f"Fixed tests: {len(fixes)}")
+    print(f"Newly skipped tests: {len(newly_skipped)}")
+    print(f"Newly passing tests (previously skipped): {len(newly_passing)}")
+
+    output_lines = []
+
+    # Report regressions
+    if real_regressions:
+        print("\nREGRESSIONS (non-intermittent failures):", file=sys.stderr)
+        for test in sorted(real_regressions):
+            msg = f"GNU test failed: {test}. {test} is passing on 'main'. Maybe you have to rebase?"
+            print(f"::error ::{msg}", file=sys.stderr)
+            output_lines.append(msg)
+
+    # Report intermittent issues
+    if intermittent_regressions:
+        print("\nINTERMITTENT ISSUES (ignored):", file=sys.stderr)
+        for test in sorted(intermittent_regressions):
+            msg = f"Skip an intermittent issue {test} (fails in this run but passes in the 'main' branch)"
+            print(f"::notice ::{msg}", file=sys.stderr)
+            output_lines.append(msg)
+
+    # Report fixes
+    if fixes:
+        print("\nFIXED TESTS:", file=sys.stderr)
+        for test in sorted(fixes):
+            msg = f"Congrats! The gnu test {test} is no longer failing!"
+            print(f"::notice ::{msg}", file=sys.stderr)
+            output_lines.append(msg)
+
+    # Report newly skipped and passing tests
+    if newly_skipped:
+        print("\nNEWLY SKIPPED TESTS:", file=sys.stderr)
+        for test in sorted(newly_skipped):
+            msg = f"Note: The gnu test {test} is now being skipped but was previously passing."
+            print(f"::warning ::{msg}", file=sys.stderr)
+            output_lines.append(msg)
+
+    if newly_passing:
+        print("\nNEWLY PASSING TESTS (previously skipped):", file=sys.stderr)
+        for test in sorted(newly_passing):
+            msg = f"Congrats! The gnu test {test} is now passing!"
+            print(f"::notice ::{msg}", file=sys.stderr)
+            output_lines.append(msg)
+
+    if args.output and output_lines:
+        with open(args.output, "w") as f:
+            for line in output_lines:
+                f.write(f"{line}\n")
+
+    # Return exit code based on whether we found regressions
+    return 1 if len(real_regressions) > 0 else 0
```

Comment:
You can simplify it to:
```suggestion
    return 1 if real_regressions else 0
```